nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -873,7 +881,9 @@ sub filter_units {
 
     if ($state->{substate} eq "auto-restart") {
         # A unit in auto-restart substate is a failure *if* it previously failed to start
-        my $main_status = `$new_systemd/bin/systemctl show --value --property=ExecMainStatus '$unit'`;
+        open(my $main_status_fd, "-|", "$new_systemd/bin/systemctl", "show", "--value", "--property=ExecMainStatus", $unit) || die("Unable to call 'systemctl show'");
+        my $main_status = do { local $/ = undef; <$main_status_fd> };
+        close($main_status_fd) || die("Unable to close 'systemctl show' fd");
```

Comment:
Why not just `my $main_status = system("$new_systemd/bin/systemctl", "show", ...);` ?